src/src/com/microsoft/aad/adal/ApplicationReceiver.java

```diff
@@ -131,45 +152,74 @@ public static void clearUserName(Context ctx) {
         }
     }
 
-    private void resumeRequestInBroker(Context ctx, String request) {
-        Logger.v(TAG, "ApplicationReceiver:resumeRequestInBroker");
+    private void resumeRequestInBroker(final Context context, final String request) 
+    {
+        final String methodName = ":resumeRequestInBroker";
+        Logger.v(TAG + methodName, "Start resuming request in broker");
         Gson gson = new Gson();
-        AuthenticationRequest pendingRequest = gson.fromJson(request, AuthenticationRequest.class);
-        Intent intent = new Intent();
-        intent.setAction(Intent.ACTION_PICK);
-        intent.putExtra(AuthenticationConstants.Broker.BROKER_REQUEST, pendingRequest);
-        intent.putExtra(AuthenticationConstants.Broker.CALLER_INFO_PACKAGE, ctx.getPackageName());
-        intent.putExtra(AuthenticationConstants.Broker.BROKER_REQUEST_RESUME,
+        final AuthenticationRequest pendingRequest = gson.fromJson(request, AuthenticationRequest.class);
+        ExecutorService sThreadExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Consider consolidating the executor service to a single instance and run a cachedThreadPool in later fixes. you an resolve this comment by filing an issue